packages-backend/express/src/auth.ts

```diff
@@ -139,6 +128,21 @@ export const getConfiguredAudience = <Request extends TBaseRequest>(
   }
 };
 
+type TJwksClient = ReturnType<typeof createRemoteJWKSet>;
+const jwksClientByIssuer = new Map<string, TJwksClient>();
+
+function getJwksClientByIssuer(issuer: string) {
+  const client = jwksClientByIssuer.get(issuer);
+  if (client) {
+    return client;
+  }
+  const newClient = createRemoteJWKSet(
+    new URL(`/.well-known/jwks.json`, issuer)
+  );
+  jwksClientByIssuer.set(issuer, newClient);
+  return newClient;
+}
```

Comment:
This is needed as there might be different issuers being used, as the issuer can also be determined dynamically by the middleware.
Therefore, we "cache" each instance of `createRemoteJWKSet` for each issuer.